File: src/v2/chromatic_circle/mixers/RedMixer.ts
import { Color } from '../Color';
import { ColorMixer } from './ColorMixer';

export class RedMixer implements ColorMixer {
  private readonly color: Color;

  constructor() {
    this.color = Color.RED;
  }

  public canMix(color: Color): boolean {
    return color === this.color;
  }

  public mixColors(color: Color): Color {
    if (color == this.color) {
      return color;
    }
    if (color === Color.YELLOW) {
      return Color.ORANGE;
    }
    if (color === Color.BLUE) {
      return Color.VIOLET;
    }
    throw new Error(`invalid color: ${color}`);
  }
}

File: java/src/main/java/com/circulocromatico/RedMixer.java
package com.circulocromatico;

public class RedMixer implements ColorMixer {
    public static final ChromaticCircle.Color RED = ChromaticCircle.Color.RED;

    @Override
    public ChromaticCircle.Color mixColor(ChromaticCircle.Color color) {
        if (RED.equals(color)) {
            return color;
        }
        if (color == ChromaticCircle.Color.YELLOW) {
            return ChromaticCircle.Color.ORANGE;
        }
        if (color == ChromaticCircle.Color.BLUE) {
            return ChromaticCircle.Color.VIOLET;
        }
        throw new IllegalArgumentException("Invalid color: " + color);
    }

    @Override
    public boolean canMix(ChromaticCircle.Color color) {
        return color == RED;
    }
}